src/fal/cli/flow_runner.py

```diff
@@ -87,11 +102,53 @@ def _flow_node_to_fal_scripts(list: List[Union[FalFlowNode, None]]) -> List[FalS
     return new_list
 
 
-def _unique_id_to_model_name(unique_id: str):
+def _unique_id_to_model_name(unique_id: str) -> str:
     split_list = unique_id.split(".")
     # if its a unique id 'model.fal_test.model_with_before_scripts'
     return split_list[len(split_list) - 1]
 
 
-def _unique_ids_to_model_names(id_list: List[str]):
+def _unique_ids_to_model_names(id_list: List[str]) -> List[str]:
     return list(map(lambda id: _unique_id_to_model_name(id), id_list))
+
+
+def _combine_fal_run_results(target_path):
+    result_files = _get_fal_run_results(target_path)
+    all_results = reduce(
+        lambda all, next: all.extend(_get_result_array(next)) or all,
+        result_files,
+        [],
+    )
+
+    last_run_result = _get_all_result_content(result_files[-1])
+    last_run_result[RUN_RESULTS_KEY] = all_results
+    run_result_file = os.path.join(target_path, RUN_RESULTS_FILE_NAME)
+
+    # remove all run_results_*.json files for the next fal flow run
+    _remove_all(result_files)
```

Comment:
nitpicking: this could not run if there was an exception in a user script and would leave the target "dirty".

Also, what `run_results.json` do we create if there is a failure.

Maybe for a different ticket/PR, but to consider.